node/src/p2p/p2p_effects.rs

```diff
@@ -290,31 +291,44 @@ pub fn node_p2p_effects<S: Service>(store: &mut Store<S>, action: P2pActionWithM
                         store.dispatch(TransitionFrontierSyncAction::BlocksPeersQuery);
                     }
                     P2pChannelsRpcAction::Timeout { peer_id, id } => {
-                        let peer = store.state().p2p.get_ready_peer(&peer_id).unwrap();
-                        let rpc_kind = peer.channels.rpc.pending_local_rpc_kind().unwrap();
-                        store.dispatch(
-                            TransitionFrontierSyncLedgerSnarkedAction::PeerQueryAddressError {
-                                peer_id,
-                                rpc_id: id,
-                                error: PeerLedgerQueryError::Timeout,
-                            },
-                        );
-                        store.dispatch(
-                            TransitionFrontierSyncLedgerStagedAction::PartsPeerFetchError {
-                                peer_id,
-                                rpc_id: id,
-                                error: PeerStagedLedgerPartsFetchError::Timeout,
-                            },
-                        );
-                        store.dispatch(TransitionFrontierSyncAction::BlocksPeerQueryError {
-                            peer_id,
-                            rpc_id: id,
-                            error: PeerBlockFetchError::Timeout,
-                        });
-                        store.dispatch(P2pDisconnectionAction::Init {
-                            peer_id,
-                            reason: P2pDisconnectionReason::TransitionFrontierRpcTimeout(rpc_kind),
-                        });
+                        if let Some(peer) = store.state().p2p.get_ready_peer(&peer_id) {
```

Comment:
what about

```rust
let Some(peer) = .... else {
    bug_condition!("get_ready_peer({:?}) returned None", peer_id);
    return;
};
```

to avoid the nesting? Diff will be smaller too